### PR TITLE
Make logs more informative, add `connected{,_at}` to `Player`

### PR DESCRIPTION
**Pull request by Drapersniper**
+ black with Line length 99 run for free :awesome:

### CODE REVIEW

File: lavalink/enums.py [hunk 1]
```diff
@@ -101,11 +101,12 @@ class NodeState(enum.Enum):
...
-    READY = 1
-    NODE_BUSY = 2
-    RECONNECTING = 3
-    DISCONNECTING = 4
+    CREATED = 0
+    CONNECTING = 1
+    READY = 2
+    NODE_BUSY = 3
+    RECONNECTING = 4
+    DISCONNECTING = 5
```
Comment (Jackenmen) [inline lavalink/enums.py]: Possible breaking change and this is a micro release
```suggestion
    CREATED = -1
    CONNECTING = 0
    READY = 1
    NODE_BUSY = 2
    RECONNECTING = 3
    DISCONNECTING = 4
```

File: lavalink/player_manager.py [hunk 2]
```diff
@@ -63,8 +67,14 @@ def __init__(self, manager: "PlayerManager", channel: discord.VoiceChannel):
...
-            f'<Player: guild="{self.channel.guild.name}" channel="{self.channel.name}",'
-            f" playing={self.is_playing} paused={self.paused} volume={self.volume}>"
+            "<Player: "
+            f"state={self.state.name}, "
+            f'guild="{self.guild.name}" ({self.guild.id}), '
+            f'channel="{self.channel.name}" ({self.channel.id}), '
+            f"playing={self.is_playing}, paused={self.paused}, volume={self.volume}, "
+            f"queue_size={len(self.queue)}, current={self.current}, "
+            f"position={self.position}, "
+            f"length={self.current.length if self.current else 0}, node={self.node}>"
```
Comment (Jackenmen) [inline lavalink/player_manager.py]:
```suggestion
            "<Player: "
            f"state={self.state.name}, "
            f"guild={self.guild.name!r} ({self.guild.id}), "
            f"channel={self.channel.name!r} ({self.channel.id}), "
            f"playing={self.is_playing}, paused={self.paused}, volume={self.volume}, "
            f"queue_size={len(self.queue)}, current={self.current}, "
            f"position={self.position}, "
            f"length={self.current.length if self.current else 0}, node={self.node!r}>"
```

File: lavalink/player_manager.py [hunk 3]
```diff
@@ -143,8 +162,8 @@ async def move_to(self, channel: discord.VoiceChannel, deafen: bool = False):
         ----------
         channel : discord.VoiceChannel
         """
-        if channel.guild != self.channel.guild:
-            raise TypeError("Cannot move to a different guild.")
+        if channel.guild != self.guild:
+            raise TypeError(f"Cannot move {self} to a different guild.")
```
Comment (Jackenmen) [inline lavalink/player_manager.py]: Optional, shows the intent more clearly.
```suggestion
            raise TypeError(f"Cannot move {self!r} to a different guild.")
```

File: lavalink/rest_api.py [hunk 4]
```diff
@@ -176,6 +176,15 @@ def __hash__(self):
...
         return hash(tuple(sorted([self.track_identifier, self.title, self.author, self.uri])))
 
+    def __repr__(self):
+        return (
+            "<Track: "
+            f"track_identifier={self.track_identifier}, "
+            f"author={self.author}, "
+            f"length={self.length}, "
+            f"is_stream={self.is_stream}, uri={self.uri}, title={self.title}>"
+        )
```
Comment (Jackenmen) [inline lavalink/rest_api.py]:
```suggestion
        return (
            "<Track: "
            f"track_identifier={self.track_identifier!r}, "
            f"author={self.author!r}, "
            f"length={self.length}, "
            f"is_stream={self.is_stream}, uri={self.uri!r}, title={self.title!r}>"
        )
```

File: lavalink/rest_api.py [hunk 5]
```diff
@@ -298,7 +307,7 @@ async def _get(self, url):
                     },
                     "tracks": [],
                 }
-            log.debug(f"Received server disconnected error when player state = {self.state}")
+            log.debug("Received server disconnected error when player state = %s", self.state)
```
Comment (Jackenmen) [inline lavalink/rest_api.py]:
```suggestion
            log.debug("Received server disconnected error when player state = %s", self.state.name)
```

File: lavalink/__init__.py [hunk 6]
```diff
@@ -1,16 +1,6 @@
...
-socket_log.setLevel(logging.INFO)
-
-ws_discord_log = logging.getLogger("red.Audio.WS.discord")
-ws_ll_log = logging.getLogger("red.Audio.WS.LLServer")
-ws_rll_log = logging.getLogger("red.Audio.WS.RLL")
-log.setLevel(logging.INFO)
-ws_discord_log.setLevel(logging.INFO)
-ws_ll_log.setLevel(logging.INFO)
-ws_rll_log.setLevel(logging.INFO)
+set_logging_level(20)
```
Comment (Jackenmen) [inline lavalink/__init__.py]: The function already has a default value so let's not introduce a magic number here.
```suggestion
set_logging_level()
```

**Review by Jackenmen — DISMISSED** [review]
Reviewed everything except for the alive changes.

File: lavalink/player_manager.py [hunk 7]
```diff
@@ -63,8 +67,14 @@ def __init__(self, manager: "PlayerManager", channel: discord.VoiceChannel):
...
     def __repr__(self):
         return (
-            f'<Player: guild="{self.channel.guild.name}" channel="{self.channel.name}",'
-            f" playing={self.is_playing} paused={self.paused} volume={self.volume}>"
+            "<Player: "
+            f"state={self.state.name}, "
+            f"guild={self.guild.name!r} ({self.guild.id}), "
+            f"channel={self.channel.name!r} ({self.channel.id}), "
+            f"playing={self.is_playing}, paused={self.paused}, volume={self.volume}, "
+            f"queue_size={len(self.queue)}, current={self.current}, "
```
Comment (Drapersniper) [inline lavalink/player_manager.py]:
```suggestion
            f"queue_size={len(self.queue)}, current={self.current!r}, "
```

File: lavalink/player_manager.py [hunk 8]
```diff
@@ -63,8 +67,14 @@ def __init__(self, manager: "PlayerManager", channel: discord.VoiceChannel):
 
     def __repr__(self):
```
Comment (Jackenmen) [inline lavalink/player_manager.py:68]: imo we should add the state of `alive` attr to the repr

Perhaps also connected_at though I'm not sure whether that attr is all that useful for the repr

File: lavalink/lavalink.py [hunk 9]
```diff
@@ -116,6 +118,7 @@ async def connect(channel: discord.VoiceChannel, deafen: bool = False):
     """
     node_ = node.get_node(channel.guild.id)
     p = await node_.player_manager.create_player(channel, deafen=deafen)
+    p.connected_at = datetime.datetime.now(datetime.timezone.utc)
```
Comment (Jackenmen) [inline lavalink/lavalink.py]: This is already handled by `Player.connect()` which is called by `PlayerManager.create_player()`. Alternatively, `PlayerManager.create_player()` may call `Player.move_to()` but I don't think we want to update this attr in such cases (and if we do, that should be done in `Player.move_to()`)

File: lavalink/player_manager.py [hunk 10]
```diff
@@ -338,20 +344,17 @@ async def play(self):
...
         self, track: Track, replace: bool = True, start: int = 0, pause: bool = False
     ):
-        log.debug(f"Resuming current track for player: {self.channel.id}.")
+        log.debug("Resuming current track for player: %r.", self)
         self._is_playing = False
         self._paused = True
-        await self.node.play(
-            self.channel.guild.id, track, start=start, replace=replace, pause=True
-        )
+        self._alive = True
```
Comment (Jackenmen) [inline lavalink/player_manager.py]:
```suggestion
```

File: lavalink/player_manager.py [hunk 11]
```diff
@@ -329,6 +334,7 @@ async def play(self):
         self.current = None
         self.position = 0
         self._paused = False
+        self._alive = True
```
Comment (Jackenmen) [inline lavalink/player_manager.py]:
```suggestion
```

File: lavalink/player_manager.py [hunk 12]
```diff
@@ -363,13 +366,14 @@ async def stop(self):
...
         """
-        await self.node.stop(self.channel.guild.id)
+        await self.node.stop(self.guild.id)
         self.queue = []
         self.current = None
         self.position = 0
         self._paused = False
         self._is_autoplaying = False
         self._auto_play_sent = False
+        self._alive = True
```
Comment (Jackenmen) [inline lavalink/player_manager.py]:
```suggestion
```

File: lavalink/player_manager.py [hunk 13]
```diff
@@ -54,6 +55,9 @@ def __init__(self, manager: "PlayerManager", channel: discord.VoiceChannel):
         self._is_autoplaying = False
         self._auto_play_sent = False
         self._volume = 100
+        self.state = PlayerState.CREATED
+        self.connected_at = datetime.datetime.now(datetime.timezone.utc)
```
Comment (Jackenmen) [inline lavalink/player_manager.py]: Should this really be set to if we haven't connected yet?

**Review by Jackenmen — CHANGES_REQUESTED** [review]

File: lavalink/lavalink.py [hunk 14]
```diff
@@ -1,4 +1,5 @@
 import asyncio
+import datetime
```
Comment (Jackenmen) [inline lavalink/lavalink.py]:
```suggestion
```

**Review by Jackenmen — CHANGES_REQUESTED** [review]

File: lavalink/__init__.py [hunk 15]
```diff
@@ -19,4 +9,37 @@
...
+    "connect",
+    "get_player",
+    "close",
+    "register_event_listener",
+    "unregister_event_listener",
+    "register_update_listener",
+    "unregister_update_listener",
+    "register_stats_listener",
+    "unregister_stats_listener",
+    "all_players",
```
Comment (Jackenmen) [inline lavalink/__init__.py:42]:
```suggestion
    "all_players",
    "all_connected_players",
```

**Review by Jackenmen — CHANGES_REQUESTED** [review]

**Review by Jackenmen — APPROVED** [review]
LGTM

**Pull request merged**